_posts/2020-02-09-SVGD.markdown

```diff
@@ -0,0 +1,401 @@
+  
+  **Optional Reading**:
+
+  1. [SVGD: Theory and Applications](https://www.cs.dartmouth.edu/~qliu/PDF/svgd_aabi2016.pdf).
+  2. [Learning to Sample with Amortized SVGD](https://arxiv.org/abs/1707.06626).
+  
+  **Questions**:
+
+  1. Compare and contrast the method shown here and MCMC. What are some advantages MCMC still has over SVGD?
+     <details><summary>Solution</summary>
+      <p>
+          Below are some ideas we discussed in our class. <br/>
+
+          <b>1.</b> SVGD requires a compact subspace \( \mathcal{X} \), and as noted <a href="http://proceedings.mlr.press/v97/chen19b/chen19b.pdf">here in Chen '19</a>, requires the number of particles to be fixed apriori.<br/>
+
+          <b>2.</b> SVGD has a lot less theoretical understanding compared to MCMC (which, is potentially due to the recency of the result). SVGD has had analysis done in the infinite-particle regime, but minimal work done in finite particle scenarios (an example of such work can be found <a href="https://papers.nips.cc/paper/8101-stein-variational-gradient-descent-as-moment-matching.pdf">here</a>. A concern of theoretical analysis is the complexity of analyzing the interacting particle updates, so the works covered here either view it from a dynamical systems / differential equation perspective (which concerns the smooth transformation of density), or discuss the properties of the final particles, regardless of how they were algorithmically attained.<br/>
+
+          <b>3.</b> SVGD still seems to collapse in high-dimensional spaces, leading to exciting new research in <a href="https://arxiv.org/abs/1902.03394">why this occurs</a> and <a href="https://arxiv.org/abs/1711.04425">ideas on how to get around it</a>.
+     </p>
+     </details>
+  2. Prove that the discrepancy in Equation 3 of the Stein Variational Gradient Descent Paper only equals 0 when \(p\) and \(q\) are equal.
+     <details><summary>Solution</summary>
+     <p>
+      Recall the operator definition of Stein's Identity:
+
+      $$ \mathbf{E}_p[\mathcal{A}_pf(x)] = 0$$
+
+      If \( p \neq q \), we get \( \mathbf{E}_q[\mathcal{A}_pf(x)] \) for some choice of function \( f \).
+
+      We can expand this to:
+
+      $$\mathbf{E}_q[\mathcal{A}_pf(x)] - \mathbf{E}_q[\mathcal{A}_qf(x)]$$
+
+      Recalling the full definition of the operator:
+
+      $$\mathcal{A}_pf(x) = \mathbf{E}_p[s_p(x)f(x) + \nabla_x f(x)] = 0$$
+
+      where score function \( s_p(x) \) is just \( \nabla_x \log p(x) \), we are left with
+
+      $$\mathbf{E}_q[(s_p(x) - s_q(x))f(x)]$$
+
+      This means unless \(p = q \rightarrow s_p(x) = s_q(x) \; \forall x \in \mathcal{X} \), we can always find some function \(f\) for which the above quantity is nonzero.
+
+     </p>
+     </details>
+  3. Implement SVGD in your favorite language (see the notes for links to different implementations). Then, let’s take a look at the role of the kernel in SVGD:
```

Comment:
Can you put this as a markdown list please? It shows up really oddly wrt spacing otherwise.